compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -2357,22 +2359,21 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
         // 2) `std` suggestions before `core` suggestions.
         let mut extern_crate_names =
             self.extern_prelude.keys().map(|ident| ident.name).collect::<Vec<_>>();
-        extern_crate_names.sort_by(|a, b| b.as_str().partial_cmp(a.as_str()).unwrap());
-
-        for name in extern_crate_names.into_iter() {
-            // Replace first ident with a crate name and check if that is valid.
-            path[0].ident.name = name;
-            let result = self.maybe_resolve_path(&path, None, parent_scope, None);
-            debug!(
-                "make_external_crate_suggestion: name={:?} path={:?} result={:?}",
-                name, path, result
-            );
-            if let PathResult::Module(..) = result {
-                return Some((path, None));
-            }
-        }
-
-        None
+        extern_crate_names.sort_by_key(|&name| Reverse(name));
+
+        extern_crate_names
```

Comment:
I feel like this is harder to read than the existing code. Can you revert it?

Reply:
Yeah, you're right, also doesn't fit the commit it was in. Reverted.

Reply:
Using `Reverse` here was actually wrong because I was ordering by `Symbol` (which orders by index) instead of its string repr. Unfortunately I had to revert to `sort_by` because of lifetime issues, but at least I could still use total ordering instead of partial.